featurebyte/feast/patch.py

```diff
@@ -101,16 +101,48 @@ def augment_response_with_on_demand_transforms(
             )
 
 
+class DataFrameWrapper(pd.DataFrame):
+    """
+    Wrapper class for pandas DataFrame to support alias column names. This feature is used to improve the
+    runtime & memory performance of the get_transformed_features_df function.
+    """
+
+    def __init__(self, *args: Any, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+        self._alias: Dict[str, str] = {}
+
+    def add_column_alias(self, column_name: str, alias: str) -> None:
+        """
+        Add a column alias
+
+        Parameters
+        ----------
+        column_name: str
+            Column name
+        alias: str
+            Alias name of the column
+        """
+        self._alias[alias] = column_name
+
+    def __getitem__(self, key: Any) -> Union[pd.Series, pd.DataFrame]:
+        if not isinstance(key, str) and isinstance(key, Iterable):
+            return pd.DataFrame({_key: self.__getitem__(_key) for _key in key})
+
+        if isinstance(key, str) and key in self._alias:
+            key = self._alias[key]
+
+        return super().__getitem__(key)
```

Comment:
By extending pandas DataFrame with alias column support, we can get rid of the expensive dataframe expansion & column dropping operations.